libs/src/services/ethContracts/serviceProviderFactoryClient.js

```diff
@@ -223,30 +223,30 @@ class ServiceProviderFactoryClient extends GovernedContractClient {
     return info
   }
 
-  async getServiceProviderIdsFromAddress (ownerAddress, serviceType) {
+  async getServiceProviderIdsFromAddress(ownerAddress, serviceType) {
     const method = await this.getMethod('getServiceProviderIdsFromAddress',
       ownerAddress,
       Utils.utf8ToHex(serviceType)
     )
     let info = await method.call()
-    return info
+    return info.map(id => parseInt(id))
   }
 
-  async getServiceProviderIdFromAddress (ownerAddress, serviceType) {
+  async getServiceProviderIdFromAddress(ownerAddress, serviceType) {
     const infos = await this.getServiceProviderIdsFromAddress(ownerAddress, serviceType)
-    return infos[0]
+    return infos[0] ? parseInt(infos[0]) : null
```

Comment:
i don't think this function is exactly valid? since there can be > 1 id per address for a given service type, maybe worth removing